bin/logstash.lib.sh

```diff
@@ -131,13 +131,19 @@ setup_ruby() {
   VENDORED_JRUBY=
 }
 
+logging_opts() {
+    printf "%s" " -J-DLog4jLogEventFactory=org.logstash.log.CustomLogEventFactory"
+    printf "%s" " -J-DLog4jContextSelector=org.logstash.log.StructuredMessageContextSelector"
```

Comment:
I used to have this, but then was worried this would conflict with other things... but I guess I override it anyways and others can do the same by added extra arguments when running LS